File: all programs/armstrong number.c
#include<stdio.h>
#include<math.h>
int main(){
    int num;
    scanf("%d",&num);
    int temp = num;
    int count =0;
    int sum =0;
    while(temp>0){
        temp = temp/10;
        count++;
    }
    temp = num;
    while(temp>0){
       int rem = temp%10;
        sum = sum + pow(rem,count);
        temp = temp/10;
    }
    if(sum==num){
        printf("it is a amstrong number");
    }
    else{
        printf("it is not a armstrong number");
    }
}

File: python/pythonlevel2/armstrongnumber.py
import math
a=int(input())
temp = a
tot = 0
count = str(a)
cnt = len(count)
# while(temp>0):
#     count = count + 1
#     temp = temp/10
# temp = a
while(temp>0):
    rem = temp%10
    re = int(rem)
    tot = tot + pow(re,cnt)
    temp = temp/10
if(tot == a):
    print("armstrong")
else:
    print("not a armstrong")


# crt version
import math
a=int(input())
temp = a
tot = 0
count = 0
while(temp>0):
    count = count + 1
    temp = temp//10
temp = a
while(temp>0):
    rem = temp%10
    tot = tot + pow(rem,count)
    temp = temp//10
if(tot == a):
    print("armstrong")
else:
    print("not a armstrong")
print(count)